### ISSUE TITLE
Near cache eviction doesn't work

### ISSUE THREAD

**Issue by nitsandaniel** [hazelcast/hazelcast#9723]
I'm using Hazelcast 3.7.4, and I have both lite and non-lite members.  

On my lite members I'm setting near-cache as follows:
```
NearCacheConfig nearCacheConfig = new NearCacheConfig(0, cacheSize, EvictionPolicy.LRU.name(), 0, true, InMemoryFormat.OBJECT);
config.getMapConfig(mapName).setNearCacheConfig(nearCacheConfig);
```
I have configured entry listener on my lite member and could verify that it is called on change. Moreover, the old and new value content is correct.

But when I'm trying to get the value from the map I'm still getting the old value. 

While debugging, I've tried to look at the map content and I could see the new value, but when I'm calling ``map.get(key)`` the old value is returned.

**Comment by nitsandaniel** [comment]
is this a known bug?

**Comment by jerrinot** [comment]
hi @nitsandaniel, it's not a known bug, not known to me anyway. I just added categorization to make sure it's addressed before 3.8 release. I see you closed the issue. is it working for you after all?

**Comment by nitsandaniel** [comment]
I accidentally closed it, reopened it immediately.

**Comment by ahmetmircik** [comment]
Hi @nitsandaniel,
`IMap` and its near cache have isolated eviction mechanisms. When an entry is evicted from `IMap`, this doesn't require an eviction on near cache or vice versa. So it can be expected that near cache returns the evicted entry because it still owns it. 

Since eviction doesn't affect correctness of the data, if the latest state of entry is still observable, returning it from near cache should not cause any problem even `IMap` doesn't have it. This makes sense to you?

**Comment by nitsandaniel** [comment]
Actually it doesn't.

Let's say that the key 'a' was mapped to the value '1', but then changed to '2', how could it make sense that I'll still get '1' when I'm querying the map for 'a'? 

I'm not talking about eviction here but value update.
Moreover, one of the near-cache constructor is ``invalidateOnChange``, which is set to ``true`` in my code.

**Comment by ahmetmircik** [comment]
> Let's say that the key 'a' was mapped to the value '1', but then changed to '2', how could it make sense that I'll still get '1' when I'm querying the map for 'a'?

Thanks for clarification. Obviously this is an issue and it is not a known one. Can you please share a reproducer?

> I'm not talking about eviction here but value update.
Ok i was describing eviction behaviour.

**Comment by nitsandaniel** [comment]
The code snippet above is how I configure near cache.
Other than that, the usage code is very simple:
```
IMap<MyKey, MyValueObject>map = hazelcastInstance.getMap(mapName);
MyValueObject value = map.get(key);
```

**Comment by ahmetmircik** [comment]
Thanks for reporting this, indeed it is a bug for 3.7 version. 3.8-EA don't have this issue.
Root cause of the issue is invalidations are not sent to lite-members if near cache is not configured also on other members.

For now, as a workaround you can enable near cache on other members. Possibly we can fix this issue in next 3.7 patch release which will be 3.7.6.

**Comment by nitsandaniel** [comment]
Are there planned timelines for 3.7.6 and 3.8?

**Comment by ahmetmircik** [comment]
ETA for both is mid february.

**Comment by jerrinot** [comment]
fixed by #9890
